cmd/deploy.go

```diff
@@ -766,6 +785,32 @@ func printDeployMessages(out io.Writer, f fn.Function) {
 	}
 }
 
+// isTagged returns true if provided image string 'v' has valid tag and false if
+// not. It is lenient in validating - does not always throw an error, just
+// returning false in some scenarios.
+func isTagged(v string) (validTag bool, err error) {
+	if strings.Contains(v, "@") {
+		// digest has been processed separately
+		return
+	}
+	vv := strings.Split(v, ":")
+	if len(vv) < 2 {
+		err = fmt.Errorf("image '%v' does not contain a tag", v)
+		return
+	} else if len(vv) > 2 {
+		err = fmt.Errorf("image '%v' contains an invalid tag (extra ':')", v)
+		return
+	}
+	tag := vv[1]
+	if tag == "" {
+		err = fmt.Errorf("image '%v' has an empty tag", v)
+		return
+	}
```

Comment:
Do you think this is an error, or should just return false?  For example, is a trailing `:` just another way to be "untagged" (i.e. the colon is superfulous)

Reply:
This was my dilemma. Do we want to ease on these restrictions and allow users to do "more" because using `--image` implies "I know what Im doing" or should we still kind of help them along? maybe even in this case it could append ":latest" at the end or something for simplified input.